internal/common/types.go

```diff
@@ -0,0 +1,126 @@
+package store
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/google/uuid"
+)
+
+// ParseCustomTypeError is thrown when parsing strings into enumerations
+type ParseCustomTypeError struct {
+	reason string
+}
+
+func (err *ParseCustomTypeError) Error() string {
+	return err.reason
+}
+
+// Architecture represents one of the supported CPU architectures available for images
+// produced by osbuild-composer. I decided to represent it as an integer because if it
+// was a string it would unmarshal from JSON just fine even in case that the architecture
+// was unknown (yeah, thanks Go for doing the same mistakes C did almost 50 years ago)
```

Comment:
I don't disagree, but drop the comments in the parenthesis to make the tone of the documentation more neutral.